components/consent-mgt/org.wso2.carbon.identity.consent.mgt/src/main/java/org/wso2/carbon/identity/consent/mgt/listener/ConsentDeletionAppMgtListener.java

```diff
@@ -135,6 +136,50 @@ public boolean doPostDeleteApplication(String applicationName, String tenantDoma
         return true;
     }
 
+    /**
+     * When an application is disabled, it will delete all relevant receipts issued against that application.
+     *
+     * @param serviceProvider Service provider which is disabled.
+     * @param tenantDomain    Tenant domain of the application.
+     * @param userName        Username of the person who does the deletion.
+     * @return true.
+     * @throws IdentityApplicationManagementException IdentityApplicationManagementException.
+     */
+    @Override
+    public boolean doPostUpdateApplication(ServiceProvider serviceProvider, String tenantDomain, String userName)
+            throws IdentityApplicationManagementException {
+
+        ConsentManager consentManager = IdentityConsentDataHolder.getInstance().getConsentManager();
+        if (serviceProvider.isApplicationEnabled()) {
```

Comment:
move this code block up. we don't need to do `ConsentManager consentManager = IdentityConsentDataHolder.getInstance().getConsentManager();` if the app is not disabled.